Sources/AppStoreConnectCLI/Files/BetaGroupFileManager.swift

```diff
@@ -0,0 +1,58 @@
+// Copyright 2020 Itty Bitty Apps Pty Ltd
+
+import Foundation
+import Files
+import Yams
+
+struct BetaGroupFolderManager: ResourceFolderManager {
+
+    let path: String
+
+    func read() throws -> [BetaGroup] {
+        var localGroups: [BetaGroup] = []
+
+        for file in try Folder(path: path).files {
+            if file.extension == "yml" {
+                let group = Readers
+                    .FileReader<BetaGroup>(format: .yaml)
+                    .read(filePath: file.path)
+
+                localGroups.append(group)
+            }
+        }
+
+        return localGroups
+    }
+
+    func save(_ betaGroups: [BetaGroup]) throws {
+        let folder = try Folder(path: "").createSubfolderIfNeeded(at: path)
+
+        try betaGroups.forEach {
+            let groupFile = try folder.createFile(named: "\($0.formattedFileName).yml")
+
+            try groupFile.write(try YAMLEncoder().encode($0))
+        }
+    }
+
+    func save(groupsWithTesters betaGroupWithTesters: [(BetaGroup, [BetaTester])]) throws {
+
+        let folder = try Folder(path: "").createSubfolderIfNeeded(at: path)
+
+        try folder.files.forEach { try $0.delete() }
+
+        let betaGroups = try betaGroupWithTesters.map { (group: BetaGroup, testers: [BetaTester]) -> BetaGroup in
+            let testersFile = try folder
+                .createFile(named: "\(group.formattedFileName)_beta-testers.csv")
+
+            try testersFile.write(testers.renderAsCSV())
+
+            var group = group
+
+            group.testers = testersFile.path
+
+            return group
+        }
```

Comment:
I can see what you were going for here, however you usually don't want to perform side effects in a `map` function. To explain more explicitly you would usually expect your `map` to be a pure transformation and not do things like write out files.

This could be cleaned up by extracting a private func `save(_ betaGroup:)` out of `save(_ betaGroups:)` and use a for each loop here to write out the testers and then the betagroup.

Now that I write this though it occurs to me that we probably want a singular file to describe all the groups? But maybe not... One file per group might be okay. @orj any thoughts?